### ISSUE TITLE
[1.2.0b2] borg info: error about missing "files" cache file and not getting actual info

### ISSUE THREAD

**Issue by FabioPedretti** [borgbackup/borg#5673]
## Is this a BUG / ISSUE report or a QUESTION?

Bug

#### Your borg version (borg -V).
```
# ./borg-1.2.0b2 -V
borg-1.2.0b2 1.2.0b2
# borg -V
borg 1.1.15
```
#### Operating system (distribution) and version.
Up to date 64 bit Debian 10:
```
# uname -a
Linux borg 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux
```
#### Hardware / network configuration, and filesystems used.
A vsphere VM, repo on SSHFS, files to backup mounted with NFS.

#### How much data is handled by borg?
See output of borg info below.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
borg info (detailed below)

## Describe the problem you're observing.

With 1.2.0b2 borg info shows an error* and get stuck until I kill it. Using a repo created with previous 1.1 release.
*Note that I am using borg 1.1.15 with BORG_FILES_CACHE_SUFFIX set to a different value for every archive, so the "files" file doesn't exist, but there are many "files.archive-number" files.
The BORG_FILES_CACHE_SUFFIX env var is set just during the create, not while doing borg info.

No issues with borg 1.1.15 (completes in about 5 seconds):
```
# borg info
Repository ID: xxx
Location: /mnt/yyy/borg-repo
Encrypted: Yes (repokey BLAKE2b)
Cache: /mnt/yyy/borg-cache/xxx
Security dir: /root/.config/borg/security/xxx
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:              570.08 TB            433.07 TB              2.45 TB

                       Unique chunks         Total chunks
Chunk index:                 5142003           1124036218
```
Error + getting stuck for some minutes (until I kill it) with borg 1.2.0b2:
```
# ./borg-1.2.0b2 info
The files cache can't be read. [[Errno 2] No such file or directory: '/mnt/yyy/borg-cache/xxx/files']
Continuing without files cache - expect lower performance.
Repository ID: xxx
Location: /mnt/yyy/borg-repo
Encrypted: Yes (repokey BLAKE2b)
Cache: /mnt/yyy/borg-cache/xxx
Security dir: /root/.config/borg/security/xxx
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
^C^C^C^CKeyboard interrupt
```

**Comment by ThomasWaldmann** [comment]
It's in `borg.cache.LocalCache._do_open`:

if it is not disabled via `--files-cache=disabled` (which is only available for `borg create`), that unconditionally loads the files cache.

**Comment by ThomasWaldmann** [comment]
Maybe it should be `constants.DEFAULT_FILES_CACHE_MODE='d'`.

**Comment by FabioPedretti** [comment]
I tested 1.2.0b3, the error is gone, however it is still very slow (I killed it after some minutes), while with borg 1.1.16 it completes in a few seconds.
Should I open a new issue?

**Comment by ThomasWaldmann** [comment]
@FabioPedretti yes, please. and maybe write a "reproducer script" with pathes relative to ~.

**Comment by FabioPedretti** [comment]
Reported at #5804.

**Comment by ThomasWaldmann** [comment]
backport to 1.1-maint: #6626

**Comment by ThomasWaldmann** [comment]
@FabioPedretti can you practically try it?

**Comment by FabioPedretti** [comment]
> @FabioPedretti can you practically try it?

I tried borg prune with 1.2.0 (which includes the original commit since 1.2.0b3) and it work as expected.
Unfortunately I cannot try the development version in 1.1, when there will be a binary available I will test it.

**Comment by ThomasWaldmann** [comment]
OK, it later will be in 1.1.18.

**Comment by FabioPedretti** [comment]
Testing 1.1.18 since its release and work as expected.
